samples/drivers/usbc/sink/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(usbc_sink)
+
+target_sources(app PRIVATE src/main.c)
+target_sources(app PRIVATE src/usbc_snk.c)
+
+if("${BOARD}" STREQUAL "stm32g071b_disco")
+  target_sources(app PRIVATE src/stm32g071b_disco_board.c)
+else()
+  target_sources(app PRIVATE src/stm32g081b_eval_board.c)
```

Comment:
those seems to be very board specific sources.

Generally we should avoid such constructs, especially because it makes it harder to reuse a sample on another board which shares same SoC or peripheral as those boards.

Is it not possible to select source file based on devicetree knowledge or Kconfig setting ?

And if not, what makes this sample so special it should be restricted to those board, and not a general restriction based on hw capabilities.
For example, like the bluetooth samples of course requires a radio peripheral, but we don't test for `n` different boards in those samples.

At this point i'm asking in order to get a better understanding.

Reply:
Yes, those files are very board specific. They implement a common API used by the sample to configure the board for USBC operation. For example, the stm32g081b eval board uses on on chip ADC to measure VBUS but the stm32g071b disco board uses an external I2C INA device to measure VBUS.

Reply:
> stm32g071b disco board uses an external I2C INA device to measure VBUS

with this, I assume you refer to the `ti,ina230` compatible device: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/sensor/ti%2Cina230.yaml

That would mean that any board / shield having the `ti,ina230` compatible should be working, correct ?

Then that should be your test, and not the board.

Reply:
Sure, but some boards might measure VBUS in some other way other than ti,ina230 or an internal ADC. Also, this was just one example, there could be several GPIOs that need configured to properly enable USBC functionality.

Reply:
>  there could be several GPIOs that need configured to properly enable USBC functionality.

Sure, but that's still not a reason to be board centric.
Check for requirements that are needed for the sample.

Just like build the can driver sample:
https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/can

when building for a board without can, for example nrf52840dk_nrf52840:
```
/projects/github/ncs/zephyr/include/device.h:96:39: error: '__device_dts_ord_DT_CHOSEN_zephyr_canbus_ORD' undeclared here (not in a function)
   96 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
```

For the TCPC sample, there could even be someone adding a shield so that the sample could work on even more boards.

If this really is a board specific sample (which I don't think from the look of it), then it ought to go to https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/boards/ instead.

Reply:
I'm sorry, but I still think I'm missing something.

In the sample, main calls the board_config() function and for:
stm32g081b_eval: board_config() configures the ADC, and a pin for VBUS discharge
stm32g071b_disco: board_config() configures three pins that enable control of the CC lines

The two specific board_config() function do entirely different things that are need for the sample to work correctly.

I'm not sure how to do that through the device tree.

Also, since the board_config() functions configure things for this sample only, I don't think they should be moved to the board directory.

Reply:
What about introducing a dedicated choice Kconfig symbol  for this sample:
USBC_CONFIGURE_VBUS
USBC_CONFIGURE_ADC
There are numerous samples that add a Kconfig file for their specific configuration needs, look at for instance  `samples/subsys/usb/mass/Kconfig`

Reply:
I moved the sample to the boards directory and limited it to the stm32g081b_eval board.

Reply:
@tejlmand when you have time, could you take another look at my changes. Thank you.